TheGiftpireStrikesBack/TheGiftpireStrikesBack/Lightsabers.swift

```diff
@@ -26,3 +29,21 @@ extension Binding {
         }
     }
 }
+
+/// Monad<Wrapped?> becomes Monad<Wrapped>?
+//Binding<Wrapped?> -> Binding<Wrapped>?
+//Binding<Optional<Value>> -> Optional<Binding<Value>>
+extension Binding {
+    func flipOptional<T>() -> Optional<Binding<T>> where Value == Optional<T> { // This is correct so far
+        guard let wrappedValue = self.wrappedValue else { return nil }
+        
+        return Binding<T>(
+            get: { wrappedValue },
```

Comment:
Again, have a look at the tests. This is a really valiant attempt, but what happens if the underlying binding is changed?

It's a great attempt though!